tldraw/apps/tldraw-logseq/src/lib/shapes/IFrameShape.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react'
+import { TLBoxShape, TLBoxShapeProps } from '@tldraw/core'
+import { HTMLContainer, TLComponentProps } from '@tldraw/react'
+import { action, computed } from 'mobx'
+import { observer } from 'mobx-react-lite'
+
+export interface IFrameShapeProps extends TLBoxShapeProps {
+  type: 'iframe'
+  url: string
+}
+
+export class IFrameShape extends TLBoxShape<IFrameShapeProps> {
+  static id = 'iframe'
+  frameRef = React.createRef<HTMLIFrameElement>()
+
+  static defaultProps: IFrameShapeProps = {
+    id: 'iframe',
+    type: 'iframe',
+    parentId: 'page',
+    point: [0, 0],
+    size: [853, 480],
+    url: '',
+  }
+
+  canEdit = true
+
+  @action onIFrameSourceChange = (url: string) => {
+    this.update({ url })
+  }
+
+  @action reload = () => {
+    this.frameRef.current.src = this.frameRef?.current?.src
```

Comment:
Ideally we don't, but the problem is that we don't know what's going on within the iFrame.
The `src` of the iFrame and the `url` of the component won't update when we navigate.
I couldn't find a proper way to handle this.

Reply:
One solution might be setting `src` to `''` immediately, and set it again to `props.src` in the next tick in `setTimeout`, I think

Reply:
Actions within the iFrame won't affect the initial value of the src, as far as I understand.
The only relevant event that will be triggered is the `onLoad`, but that's the only info that we have.
Also, some actions might not even trigger a url or a history change (e.g. certain single page apps), so allowing the refresh action might be required, even if we find a way to track the url.

Reply:
Yeah, iframe is a sandbox and we have cannot know what is happening instead except:
- that site is provided by us so that we can using some custom protocols
- I am not sure if it is possible to hack the frame browser context in iframe in an Electron app?

Reply:
> Actions within the iFrame won't affect the initial value of the src, as far as I understand.
I just realized that clicking the fake reload will always reset the view. 😅